apps/protocol-frontend/src/components/RecentContributionsTable.tsx

```diff
@@ -74,9 +75,26 @@ const columnsDef: ColumnDef<UIContribution>[] = [
   },
   {
     header: 'Contributor',
-    accessorFn: contribution => contribution.user.name,
-    cell: ({ getValue }: { getValue: Getter<string> }) => {
-      return <Text>{getValue()} </Text>;
+    accessorKey: 'user',
+
+    cell: ({ getValue }: { getValue: Getter<UIContribution['user']> }) => {
+      const value = getValue();
+
+      const hasMemberName = value.name || value.display_name;
+      const displayMemberName =
+        value.name || value.display_name || displayAddress(value.address);
+      return hasMemberName ? (
+        <Tooltip
+          variant="primary"
+          label={value.address}
+          fontSize="sm"
+          placement="right"
+        >
+          <Text whiteSpace="normal">{displayMemberName}</Text>
+        </Tooltip>
+      ) : (
+        <Text whiteSpace="normal">{displayMemberName}</Text>
+      );
```

Comment:
Should we abstract this logic into a component because it is being used in three places?

Reply:
Good idea, I think it's time to do that at this point 👍 
Had a similar thought about the emojis that we're using in the CTA and Showcase (basically creating an emoji component similar to how Icon components work -- it can handle the span and aria-labelledby)

Reply:
Created a `MemberDisplayName` component that handles the logic and display for this now.

We have 1 more instance of a _similar_ logic that has additional rules (in the Member Settings table) so I left that one for now, but this replaces the logic in the Attestations/My Attestations/Recent Contributions tables.